### PR TITLE
hide file upload component on public forms

### PR DESCRIPTION
**Pull request by TimCsaky**
# Description

supports [https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1803](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1803)

- hide file upload component from list of available components in form builder.. for public forms.
if a form already contains an uploader and switches to become public, the component remains in the form but wont work because file upload requires an authenticated user's oidc bearer token with upload request. 

- the file upload component was developed by Jason in PR https://github.com/bcgov/common-hosted-form-service/pull/95

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by jujaga — APPROVED** [review]

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-123

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-123

File: openshift/README.md [hunk 1]
```diff
@@ -57,7 +57,13 @@ oc create -n $NAMESPACE configmap $APP_NAME-server-config \
   --from-literal=SERVER_PORT=8080
 ```
 
+*Note:* We use the NRS object storage for CHEFS. For related information, see:
+
+- Environment variables created by the following config map can be overridden by your [Object Storage Configuration](../app/config/test.json)
```
Comment (jujaga) [inline openshift/README.md:62]: Not sure if there's any value linking to the test.json as that specific file is only used in CI/CD environments. Instead, there should be a note saying create `local.json` and add in your values there or something along those lines.

File: app/README.md [hunk 2]
```diff
@@ -8,7 +8,8 @@ This app will require some configuration. The API will be locked down and requir
 
 ## Super Quickstart
 
-Ensure that you have filled in all the appropriate configurations following [config/custom-environment-variables.json](config/custom-environment-variables.json) before proceeding.
+Ensure that you have filled in all the appropriate configurations shown in [config/custom-environment-variables.json](config/custom-environment-variables.json) before proceeding.
+You can configure your **database connection**, **front and back-end paths** and **authentication parameters** as well as **object storage** for file uploads with your [local environment files](config/test.json) or [openshift config maps](../openshift/README.md).
```
Comment (jujaga) [inline app/README.md:12]: Not sure if there's any value linking to the test.json as that specific file is only used in CI/CD environments. Instead, there should be a note saying create `local.json` and add in your values there or something along those lines.

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-123

**Pull request merged**